python/nav/models/manage.py

```diff
@@ -303,85 +310,34 @@ def device(self):
-        warnings.warn(
-            "The Netbox.read_write attribute will be removed in a future release",
-            category=DeprecationWarning,
-            stacklevel=2,
-        )
-        return self._get_snmp_config('community', writeable=True)
-
-    @property
-    def snmp_version(self):
-        """Returns the configured SNMP version as an integer"""
-        warnings.warn(
-            "The Netbox.snmp_version attribute will be removed in the next "
-            "feature release",
-            category=DeprecationWarning,
-            stacklevel=2,
-        )
-        value = self._get_snmp_config('version')
-        if value or value == 0:
-            if value == "2c":
-                return 2
-            return int(value)
-
-    def _get_snmp_config(self, variable='community', writeable=None):
-        """Returns SNMP profile configuration variables, preferring the profile
-        with the highest available SNMP version.
-        """
-        # TODO: This method can be removed when the SNMP properties above are removed
-        query = Q(protocol=ManagementProfile.PROTOCOL_SNMP)
-        if writeable:
-            query = query & Q(configuration__write=True)
-        elif writeable is not None:
-            query = query & (
-                Q(configuration__write=False) | ~Q(configuration__has_key='write')
-            )
-        profiles = sorted(
-            self.profiles.filter(query),
-            key=lambda p: str(p.configuration.get('version') or 0),
-            reverse=True,
-        )
-        if profiles:
-            return profiles[0].configuration.get(variable)
-
     def get_preferred_snmp_management_profile(
-        self, writeable=None
+        self, require_write=False
     ) -> Optional[ManagementProfile]:
-        """
-        Returns the snmp management profile with the highest available
-        SNMP version.
+        """Returns the snmp management profile with the highest available SNMP version.
+
+        :param require_write: If True, only write-enabled profiles will be
+                              considered.  If false, read-only profiles will be
+                              preferred, unless a write-enabled profile is the only
+                              available alternative.
         """
         query = Q(
             protocol__in=(
                 ManagementProfile.PROTOCOL_SNMP,
                 ManagementProfile.PROTOCOL_SNMPV3,
             )
         )
-        if writeable:
+        if require_write:
             query = query & Q(configuration__write=True)
-        elif writeable is not None:
-            query = query & (
-                Q(configuration__write=False) | ~Q(configuration__has_key='write')
+
+        profiles = self.profiles.filter(query)
+
+        if not require_write:
+            # Sort read-only profiles first
+            profiles = sorted(
+                profiles, key=lambda p: p.configuration.get("write", False)
             )
```

Comment:
What a disgusting conditional. What happens if `writeable` *is* None?

Reply:
Github is stupid again. I marked the code that was replaced by this code, which explicitly did *not* cover `writeable` being `None`. The *new* code handles all falsey situations.

Reply:
This is exactly the logic the new code is meant to clean up and clarify :)